tcms/core/management/commands/refresh_permissions.py

```diff
@@ -0,0 +1,115 @@
+from django.apps import apps
+from django.contrib.auth.models import Group, Permission
+from django.core.management import call_command
+from django.core.management.base import BaseCommand
+
+
+class Command(BaseCommand):
+    help = ('Refresh permissions for Tester group  and remove '
+            'stale ones. Use -v 0 to suppress most of the '
+            'console output'
+            )
+
+    @staticmethod
+    def get_all_permissions():
+        """
+            Returns a set of all available permissions in format
+            app_label.perm_codename
+        """
+        permissions = list()
+        for perm in Permission.objects.all():
+            permissions.append("%s.%s" %
+                               (perm.content_type.app_label, perm.codename))
+        return set(permissions)
+
+    @staticmethod
+    def get_all_installed_apps():
+        """
+            Returns a set of currently installed applications as app_labels
+        """
+        applications = list()
+        for app_config in apps.get_app_configs():
+            applications.append(app_config.label)
+        return set(applications)
+
+    @staticmethod
+    def kiwi_apps():
+        """
+            Returns a list of Kiwi TCMS related applications with permissions
+            to be assigned to Tester group
+        """
+        kiwi_apps = ['bugs', 'django_comments', 'linkreference', 'management',
+                     'testcases', 'testplans', 'testruns', 'attachments']
+        return kiwi_apps
+
+    @staticmethod
+    def permissions_exclude():
+        """
+            Returns a set of permissions NOT to be assigned to Tester group
+        """
+        perm_exclude = {'attachments.delete_foreign_attachments', }
+        return perm_exclude
+
+    @staticmethod
+    def assign_default_group_permissions(output=None, assign_if_none=False):
+        """
+            Assigns default permissions for Kiwi TCMS apps to Tester group
+        """
+        if assign_if_none:
+            admin = Group.objects.get(name='Administrator')
+            if admin.permissions.count() == 0:
+                all_perms = Permission.objects.all()
+                admin.permissions.add(*all_perms)
+
+        tester = Group.objects.get(name='Tester')
+
+        for perm in Command.get_all_permissions().difference(
+                Command.permissions_exclude()):
+            app, permission = perm.split('.')
+            if app in Command.kiwi_apps():
```

Comment:
the entire for-if block is prone to future errors and not really needed. The original variant is working just fine and needs a minor tweak in order to always update the permissions, not only when there are none set.  Return back to the original implementaiton.